structuretoolkit/analyse/snap.py

```diff
@@ -0,0 +1,397 @@
+        v=None,
+        image=None,
+        shrinkexceed=False,
+    )
+
+
+def extract_compute_np(lmp, name, compute_type, result_type, array_shape):
+    """
+    Convert a lammps compute to a numpy array.
+    Assumes the compute returns a floating point numbers.
+    Note that the result is a view into the original memory.
+    If the result type is 0 (scalar) then conversion to numpy is skipped and a python float is returned.
+    """
+    ptr = lmp.extract_compute(
+        name, compute_type, result_type
+    )  # 1,2: Style (1) is per-atom compute, returns array type (2).
+    if result_type == 0:
+        return ptr  # No casting needed, lammps.py already works
+    if result_type == 2:
+        ptr = ptr.contents
+    total_size = int(np.prod(array_shape))
+    buffer_ptr = cast(ptr, POINTER(c_double * total_size))
+    array_np = np.frombuffer(buffer_ptr.contents, dtype=float)
+    array_np.shape = array_shape
+    return array_np.copy()
+
+
+def reset_lmp(lmp):
+    lmp.command("clear")
+    lmp.command("units metal")
+    lmp.command("dimension 3")
+    lmp.command("boundary p p p")
+    lmp.command("atom_style charge")
+    lmp.command("atom_modify map array sort 0 2.0")
+
+
+def set_potential_lmp(lmp, cutoff=10.0):
+    lmp.command("pair_style zero " + str(cutoff))
+    lmp.command("pair_coeff * *")
+    lmp.command("mass * 1.0e-20")
+    lmp.command("neighbor 1.0e-20 nsq")
+    lmp.command("neigh_modify one 10000")
+
+
+def get_snap_descriptor_names(twojmax):
+    lst = []
+    for j1 in range(0, twojmax + 1):
+        for j2 in range(0, j1 + 1):
+            for j in range(j1 - j2, min(twojmax, j1 + j2) + 1, 2):
+                if j >= j1:
+                    lst.append([j1 / 2.0, j2 / 2.0, j / 2.0])
+    return lst
+
+
+def set_compute_lammps(lmp, bispec_options, numtypes):
+    lmp_variable_args = {
+        k: bispec_options[k] for k in ["rcutfac", "rfac0", "rmin0", "twojmax"]
+    }
+    lmp_variable_args.update(
+        {
+            (k + str(i + 1)): bispec_options[k][i]
+            for k in ["wj", "radelem"]
+            for i, v in enumerate(bispec_options[k])
+        }
+    )
+
+    for k, v in lmp_variable_args.items():
+        lmp.command(f"variable {k} equal {v}")
+
+    base_b = "compute b all sna/atom ${rcutfac} ${rfac0} ${twojmax}"
+    radelem = " ".join([f"${{radelem{i}}}" for i in range(1, numtypes + 1)])
+    wj = " ".join([f"${{wj{i}}}" for i in range(1, numtypes + 1)])
+    kw_options = {
+        k: bispec_options[v]
+        for k, v in {
+            "rmin0": "rmin0",
+            "bzeroflag": "bzeroflag",
+            "quadraticflag": "quadraticflag",
+            "switchflag": "switchflag",
+            "chem": "chemflag",
+            "bnormflag": "bnormflag",
+            "wselfallflag": "wselfallflag",
+            "bikflag": "bikflag",
+            "switchinnerflag": "switchinnerflag",
+            "sinner": "sinner",
+            "dinner": "dinner",
+        }.items()
```

Comment:
Is this not just `for k in [...]` since all the keys and values are equal?  I also see that those (and subsets of them) options are used later.  It'd be great to define them somewhere outside of the functions and give them names, because I don't follow at all right now.

Reply:
Done.